bokehjs/src/coffee/models/tools/draw/edit_tool.ts

```diff
@@ -0,0 +1,52 @@
+import {SelectToolView} from "../gestures/select_tool"
+
+export interface BkEv {
+  bokeh: {
+    sx: number
+    sy: number
+  }
+  srcEvent: {
+    shiftKey?: boolean
+  }
+  keyCode: number
+  timeStamp: number
+}
+
+export abstract class EditToolView extends SelectToolView {
+
+  // this is executed when the pan/drag event starts
+  _map_drag(sx: number[], sy: number[]): [number, number] {
+    const frame = this.plot_model.frame;
+
+    if (!frame.bbox.contains(sx, sy)) {
+      return
+    }
+
+    const x = frame.xscales['default'].invert(sx);
+    const y = frame.yscales['default'].invert(sy);
+    return [x, y];
+  }
+
+  _select_event(e: BkEv, append: boolean): boolean {
+    let did_hit;
```

Comment:
Every variable declaration must have type annotation. Otherwise compiler will complain with "variable has implicit type any", which indeed it currently has. In some cases this can be correctly inferred, but I still prefer to have either a declaration with type annotation or a definition (immediate assignment). Though one thing to keep in mind, is that even if a value is immediately assigned, the inferred type may be incorrect (e.g. `let x = [1, 2]` will be inferred to `number[]`, but you may want `[number, number]`).